### ISSUE TITLE
Named default channels

### ISSUE THREAD

**Issue by jakirkham** [conda/constructor#42]
Currently one can provide the option `conda_default_channels` in the `construct.yaml` file to specify what channels the installer will call `defaults` and pre-include. However, it would be nice if the default channels were not lumped into one channel called `defaults`. For instance, it would be nice to create an installer where the canonical channels in `defaults` would be included as `defaults` then `conda-forge` would be included as a default channel, but still listed as `conda-forge`. This is essential for debugging of installs to not get too hairy. If this is already provided that is awesome, please help me find the relevant info to get started using this.

If this is not already provided, I would like to propose the follow syntax. To name a channel or group of channels that is preincluded a dictionary should be used. This doesn't break the normal listing usage shown below.

``` yaml
conda_default_channels:
   - http://repo.continuum.io/pkgs/free
```

Simply one adds the dictionary key to the channel they want to name. With the example below, there is a pre-included `conda-forge` channel that pulls from `http://conda.anaconda.org/conda-forge` and a `defaults` channel that pulls from `http://repo.continuum.io/pkgs/free`.

``` yaml
conda_default_channels:
   - conda-forge: http://conda.anaconda.org/conda-forge
   - http://repo.continuum.io/pkgs/free
```

If there are multiple channels that share a name in that dictionary, a list should be used. In this case, the `conda-forge` channel pulls from `http://conda.anaconda.org/conda-forge` and the fictitious `http://conda.anaconda.org/conda-forge-test`. 

``` yaml
conda_default_channels:
   - conda-forge:
      - http://conda.anaconda.org/conda-forge
      - http://conda.anaconda.org/conda-forge-test
   - http://repo.continuum.io/pkgs/free
```

Providing this functionality of named default channels allows one to ensure the user still gets packages from the expected locations, but avoids potential issues where the definition of what is pre-included in those channels (e.g. `defaults`) breaks with existing user expectation. It also helps one to debug when packages coming from to channels that conflict with each other.

cc @loriab @msarahan

**Comment by jakirkham** [comment]
Any thoughts on this proposal?

**Comment by jakirkham** [comment]
Friendly nudge. FYI this is blocking our use of `constructor` at `conda-forge`.

**Comment by megies** [comment]
👍 sounds like a reasonable improvement.. tried to check this on a linux install, but.. where is the list of channels even stored by the installer? at least it doesn't seem to override an existing channel list in  `~/.condarc`..?

**Comment by jakirkham** [comment]
I'm not sure where this is stored either. Does anyone else have ideas?

**Comment by nehaljwani** [comment]
This is where 'defaults' comes from: https://github.com/conda/conda/blob/4.3.x/conda/base/constants.py#L63-L76

**Comment by nehaljwani** [comment]
If you look at the branch cross_conda_interface (lots of updates) there is an option called `write_condarc` which can write the defaults channels to `$INSTDIR/.condarc` https://github.com/conda/constructor/blob/cross_conda_interface/constructor/construct.py#L131-L134

**Comment by jakirkham** [comment]
Right, I knew that `conda` defined `defaults`. Was just worried about confounding this with the `defaults` that `constructor` defines. For example, if we add `conda-forge` to `conda_default_channels` and install `conda`, we would end up with `defaults` including `conda-forge` (correct?), which would make debugging much harder.

Sounds like writing `.condarc` might be a good way to go. What state is that branch in currently?

**Comment by nehaljwani** [comment]
cross_conda_interface has been merged into master and released as part of 2.0.0.

**Comment by jakirkham** [comment]
Yeah I expect you are right that `write_condarc` would have the desired result without this proposed feature. Going to close this out.
